src/herder/Upgrades.cpp

```diff
@@ -1300,8 +1300,7 @@ ConfigUpgradeSetFrameConstPtr
 ConfigUpgradeSetFrame::makeFromKey(AbstractLedgerTxn& ltx,
                                    ConfigUpgradeSetKey const& key)
 {
-    auto ltxe = ltx.loadWithoutRecord(ConfigUpgradeSetFrame::getLedgerKey(key),
-                                      /*loadExpiredEntry=*/false);
+    auto ltxe = ltx.loadWithoutRecord(ConfigUpgradeSetFrame::getLedgerKey(key));
```

Comment:
This is interesting... we don't check if the entry is live here, so we can technically upgrade using an expired entry. Should we keep this as is for more flexibility? Or should we make sure the entry is live? Checking that it's live sounds more consistent. cc @dmkozh.

Reply:
I don't have a great understanding of the upgrade system, but allowing anything to use an expired entry seems very sketchy (except for restoreOp of course). I think we should add a `isLive` check here.

Reply:
Yeah, I don't think we should need exceptions to the rules.